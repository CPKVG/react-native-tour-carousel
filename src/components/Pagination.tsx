import React from 'react';
import { StyleSheet, View } from 'react-native';
import AnimatedDotsCarousel from 'react-native-animated-dots-carousel';
import type { PaginationType } from '../types/types';

const Pagination = ({
  currentIndex,
  length,
  activeIndicatorConfig,
  inactiveIndicatorConfig,
  paginationContainerStyle,
  decreasingDots,
  maxIndicators,
}: PaginationType): JSX.Element => {
  return (
    <View
      style={paginationContainerStyle || styles.paginationContainerStyle}
      testID="pagination-container"
    >
      <AnimatedDotsCarousel
        length={length || 0}
        currentIndex={currentIndex || 0}
        maxIndicators={maxIndicators || 10}
        interpolateOpacityAndColor={true}
        activeIndicatorConfig={
          activeIndicatorConfig || {
            color: 'red',
            margin: 6,
            opacity: 1,
            size: 12,
          }
        }
        inactiveIndicatorConfig={
          inactiveIndicatorConfig || {
            color: 'grey',
            margin: 6,
            opacity: 0.4,
            size: 12,
          }
        }
        decreasingDots={
          decreasingDots || [
            {
              config: { color: 'white', margin: 3, opacity: 0.5, size: 6 },
              quantity: 1,
            },
            {
              config: { color: 'white', margin: 3, opacity: 0.5, size: 4 },
              quantity: 1,
            },
          ]
        }
      />
    </View>
  );
};
export const styles = StyleSheet.create({
  paginationContainerStyle: {
    paddingVertical: 8,
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: 20,
    height: 'auto',
  },
});

export default Pagination;
