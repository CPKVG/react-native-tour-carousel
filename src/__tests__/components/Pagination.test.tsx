import { render } from '@testing-library/react-native';
import React from 'react';
import Pagination from '../../components/Pagination';

describe('<Pagination />', () => {
  it('should handle negative currentIndex', () => {
    const { getByTestId } = render(<Pagination currentIndex={-1} length={5} />);
    const pagination = getByTestId('pagination-container');
    expect(pagination).toBeTruthy();
  });

  it('renders for a Pagination correctly', () => {
    const Renderer = render(<Pagination currentIndex={-1} length={5} />);
    expect(Renderer.toJSON()).toMatchSnapshot();
  });
});
