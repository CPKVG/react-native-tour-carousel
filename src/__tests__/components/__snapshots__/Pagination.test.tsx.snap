// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders for a Pagination correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "height": "auto",
      "justifyContent": "center",
      "marginVertical": 20,
      "paddingVertical": 8,
    }
  }
  testID="pagination-container"
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(128, 128, 128, 1)",
          "borderColor": undefined,
          "borderRadius": 12,
          "borderWidth": undefined,
          "height": 12,
          "marginHorizontal": 6,
          "marginVertical": 0,
          "opacity": 0.4,
          "width": 12,
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(128, 128, 128, 1)",
          "borderColor": undefined,
          "borderRadius": 12,
          "borderWidth": undefined,
          "height": 12,
          "marginHorizontal": 6,
          "marginVertical": 0,
          "opacity": 0.4,
          "width": 12,
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(128, 128, 128, 1)",
          "borderColor": undefined,
          "borderRadius": 12,
          "borderWidth": undefined,
          "height": 12,
          "marginHorizontal": 6,
          "marginVertical": 0,
          "opacity": 0.4,
          "width": 12,
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(128, 128, 128, 1)",
          "borderColor": undefined,
          "borderRadius": 12,
          "borderWidth": undefined,
          "height": 12,
          "marginHorizontal": 6,
          "marginVertical": 0,
          "opacity": 0.4,
          "width": 12,
        }
      }
    />
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(128, 128, 128, 1)",
          "borderColor": undefined,
          "borderRadius": 12,
          "borderWidth": undefined,
          "height": 12,
          "marginHorizontal": 6,
          "marginVertical": 0,
          "opacity": 0.4,
          "width": 12,
        }
      }
    />
  </View>
</View>
`;
